lib/benchee/statistics/percentile.ex

```diff
@@ -64,24 +59,22 @@ defmodule Benchee.Statistics.Percentile do
     percentile_value(sorted_samples, rank)
   end
 
-  defp percentile_value(sorted, rank) when trunc(rank) == 0 do
-    sorted
-    |> hd
-    |> to_float
-  end
-
-  defp percentile_value(sorted, rank) when trunc(rank) >= length(sorted) do
-    sorted
-    |> Enum.reverse
-    |> hd
-    |> to_float
-  end
-
-  defp percentile_value(sorted, rank) do
-    index = trunc(rank)
-    [lower_bound, upper_bound | _] = Enum.drop(sorted, index - 1)
-    interpolation_value = interpolation_value(lower_bound, upper_bound, rank)
-    lower_bound + interpolation_value
+  defp percentile_value(sorted_samples, rank) do
+    index = max(0, trunc(rank) - 1)
+    case Enum.split(sorted_samples, index) do
+      {_, [lower_bound, upper_bound | _]} ->
+        interpolation = interpolation_value(lower_bound, upper_bound, rank)
+        lower_bound + interpolation
+      {_, [lower_bound]} ->
+        to_float(lower_bound)
+      {[], []} ->
+        raise ArgumentError, "can't calculate percentile value on an empty list"
+      {previous_values, []} ->
+        previous_values
+        |> Enum.reverse
+        |> hd
+        |> to_float
+    end
```

Comment:
Method is a bit too long for my sensibilities :D I feel like at least the last case and maybe the first one could go into a separate function (no huge gains for the first). Wdyt?

(not totally opposed to shipping as is)